java/runtime/src/main/java/org/ray/runtime/AbstractRayRuntime.java

```diff
@@ -182,84 +163,19 @@ protected void resetLibraryPath() {
-          rayletClient.fetchOrReconstruct(batch, fetchOnly, workerContext.getCurrentTaskId());
-        }
-
-        // Get the objects from the object store, and parse the result.
-        List<GetResult<T>> getResults = objectStoreProxy.get(unreadyIds, GET_TIMEOUT_MS);
-        for (int i = 0; i < getResults.size(); i++) {
-          GetResult<T> getResult = getResults.get(i);
-          if (getResult.exists) {
-            if (getResult.exception != null) {
-              // If the result is an exception, throw it.
-              throw getResult.exception;
-            } else {
-              // Set the result to the return list, and remove it from the unready map.
-              ObjectId id = unreadyIds.get(i);
-              ret.set(unready.get(id), getResult.object);
-              unready.remove(id);
-            }
-          }
-        }
-
-        numAttempts += 1;
-        if (LOGGER.isWarnEnabled() && numAttempts % WARN_PER_NUM_ATTEMPTS == 0) {
-          // Print a warning if we've attempted too many times, but some objects are still
-          // unavailable.
-          List<ObjectId> idsToPrint = new ArrayList<>(unready.keySet());
-          if (idsToPrint.size() > MAX_IDS_TO_PRINT_IN_WARNING) {
-            idsToPrint = idsToPrint.subList(0, MAX_IDS_TO_PRINT_IN_WARNING);
-          }
-          String ids = idsToPrint.stream().map(ObjectId::toString)
-              .collect(Collectors.joining(", "));
-          if (idsToPrint.size() < unready.size()) {
-            ids += ", etc";
-          }
-          String msg = String.format("Attempted %d times to reconstruct objects,"
-                  + " but some objects are still unavailable. If this message continues to print,"
-                  + " it may indicate that object's creating task is hanging, or something wrong"
-                  + " happened in raylet backend. %d object(s) pending: %s.", numAttempts,
-              unreadyIds.size(), ids);
-          LOGGER.warn(msg);
-        }
-      }
-
-      if (LOGGER.isDebugEnabled()) {
-        LOGGER.debug("Got objects {} for task {}.", Arrays.toString(objectIds.toArray()),
-            workerContext.getCurrentTaskId());
-      }
-
-      return ret;
-    } finally {
-      // If there were objects that we weren't able to get locally, let the raylet backend
-      // know that we're now unblocked.
-      if (wasBlocked) {
-        rayletClient.notifyUnblocked(workerContext.getCurrentTaskId());
-      }
+    List<GetResult<T>> results = objectStoreProxy.get(objectIds, -1);
+    // Check exceptions before Preconditions.checkState(result.exists)
+    Optional<RayException> exception =
+        results.stream().filter(result -> result.exception != null)
+            .map(result -> result.exception).findFirst();
+    if (exception.isPresent()) {
+      throw exception.get();
     }
+    return results.stream().map(result -> {
+          Preconditions.checkState(result.exists, "The result doesn't exist.");
```

Comment:
`result.exists` shouldn't be false, as the core worker object interface will block until getting all objects, right? we can remove it.

Reply:
Yes. It's just like `RAY_CHECK`, and it shouldn't fail. However, if there's any bug and we don't have this check, we may encounter `NullPointerException`, which is hard to find out the root cause.

Reply:
Sorry, I didn't make myself clear. I mean, we can just remove the `exists` field from `GetResult` and check either data or metadata is present in `ObjectStoreProxy`.

Reply:
The `ObjectStoreProxy.get` has a timeout parameter, which means it's allowed to be inexistent. So we can't place the check inside `ObjectStoreProxy`, unless you want to check it only if the timeout parameter is negative.

Reply:
Another approach is to remove the timeout parameter of `ObjectStoreProxy.get`. We always pass the timeout value of `-1` to core worker. Then we can get rid of `GetResult` and do the check and throw the exception inside `ObjectStoreProxy.get`.

Reply:
I think we should remove the timeout parameter in the core worker `Get()` entirely actually. The only place it is actually used is in the `local_plasma_store_provider.cc`, which I don't think should be a standalone implementation of the store provider interface anyways. We can discuss that later but for now removing `ObjectStoreProxy.get` and passing in -1 to the core worker sounds like a good intermediate solution.

Reply:
One thing to note is that even without a timeout `Get()` can return `nullptr`'s if we encounter a result that was an exception and terminate early. Seems like this is handled by `result.exists`, so make sure we don't crash in that case.

Reply:
@edoakes Actually exceptions are handled by `result.exception`. Thanks for your reminder, though!

@raulchen Do you agree with removing the timeout parameter in Java code?

Reply:
I see. In the event of encountering an exception, other objects will be returned as inexistent. so `exists` should still be needed.

Reply:
And removing the `timeout` parameter from `ObjectStoreProxy.get` makes sense to me.

Reply:
@edoakes There's a unit test in Java which passes a timeout value of 0 to core worker. Removing the timeout parameter in the core worker will break it. https://github.com/ray-project/ray/blob/25b5bd15308698438bc0b857bc76568c98759bb2/java/test/src/main/java/org/ray/api/test/PlasmaFreeTest.java#L27